src/virtual_console/xlFile.c

```diff
@@ -0,0 +1,207 @@
+#include "xlFile.h"
+#include "xlHeap.h"
+#include "xlObject.h"
+
+void *func_800B0DF0(void*, size_t, s32);
+
+s32 func_800FEFF0(u32 content, u8_hdr_t **arg1, unk_file_struct *arg2);         // contentInitHandleNAND
+u32 func_800FF1B0(u8_archive_t*, char *fn, file_info_0x38_t*);                  // ARCGetFile
+s32 func_800FF28C(file_info_0x38_t*);                                           // contentGetLengthNAND
+s32 func_800FF2FC(file_info_0x38_t *arg0, char *data, u32 size, u32 pos);       // contentReadNAND
+s32 func_800FF42C(u8_archive_t*);                                               // contentReleaseHandleNAND
+s32 func_800FF424(void*);                                                       // contentCloseNAND
+
+extern class_t gClassFile;
+
+extern u8_archive_t lbl_801C9680;
+extern unk_file_struct lbl_801C96A8;
+
+extern u32 (*lbl_8025D1B0)(char*, file_info_0x38_t*);
+extern void (*lbl_8025D1B4)(list_type_t*, char*, u32, u32, u32);
+
+s32 xlFileSetOpen(void) {
+    return 1;
+}
+
+s32 xlFileSetRead(void) {
+    return 1;
+}
+
+s32 xlFileOpen(file_class_t **file, s32 arg1, char *fn) {
+    s32 ret;
+    u32 uVar1;
+
+    if (xlObjectMake((void**)file, NULL, &gClassFile) == 0) {
```

Comment:
As far as I have been able to tell basically any function that homeboy wrote return booleans.
```suggestion
    if (!xlObjectMake((void**)file, NULL, &gClassFile)) {
```